### PR TITLE
add SonataBlockBundle

### PR DESCRIPTION
**Pull request by aLkRicha**
| Q             | A
| ------------- | ---
| License       | MIT

### CODE REVIEW

File: sonata-project/block-bundle/3.9/config/packages/sonata_block.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+framework:
+    templating:
+        engines: ['twig']
+sonata_block:
+    default_contexts: [cms]
```
Comment (covex-nn) [inline sonata-project/block-bundle/3.9/config/packages/sonata_block.yaml]: `cms` context is never used anywhere, and it was renamed to `sonata_page_bundle` for SonataPageBundle (see https://stackoverflow.com/questions/22798670/what-is-context-in-sonata-block-bundle). So, i think `cms` should be removed from `default_contexts` here.

**Comment by OskarStark** [comment]
//cc @greg0ire

**Comment by aLkRicha** [comment]
i need a help with this:
```
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  
!!  In ContainerBuilder.php line 1018:
!!                                                       
!!    You have requested a non-existent service "twig".  
!!                                                       
!!  
!!  
```

it there another way then require twig-bundle in composer and enable twig service for framework?

File: sonata-project/block-bundle/3.9/config/packages/sonata_block.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+sonata_block:
+    default_contexts: [sonata_page_bundle]
```
Comment (OskarStark) [inline sonata-project/block-bundle/3.9/config/packages/sonata_block.yaml]: default_context or default_contextS?

**Comment by Nyholm** [comment]
If the bundle depends on TwigBundle it should be in the requirements section of composer.json. Or handle the case where twig does not exit. 

It seams like the recipe is good but the bundles need to be updated.

File: sonata-project/block-bundle/3.9/config/packages/sonata_block.yaml (hunk 2 above)
Comment (covex-nn) [reply sonata-project/block-bundle/3.9/config/packages/sonata_block.yaml]: @OskarStark it is `default_contexts`. See [Configuration.php](https://github.com/sonata-project/SonataBlockBundle/blob/3.x/src/DependencyInjection/Configuration.php#L84). It is required, but if the value is left blank (`default_contexts: []`), there is no errors occured, at least when i use only SonataAdminBundle

May be `default_contexts` should not be required? `sonata_page_bundle` could be appended with recipe for SonataPageBundle!

**Comment by aLkRicha** [comment]
@OskarStark @greg0ire so, we need to add TwigBundle to SonataBlockBundle?

**Comment by greg0ire** [comment]
It probably should, since we are referencing it: https://github.com/sonata-project/SonataBlockBundle/search?utf8=%E2%9C%93&q=twigbundle&type=

**Comment by aLkRicha** [comment]
@greg0ire ok, i'll create pull-request

**Comment by OskarStark** [comment]
We release [3.92](https://github.com/sonata-project/SonataBlockBundle/releases/tag/3.9.2) of block-bundle

**Comment by aLkRicha** [comment]
i think this will work after merging this https://github.com/symfony/recipes-contrib/pull/151

**Comment by covex-nn** [comment]
@aLkRicha please move `config/packages/sonata_projects/sonata_block.yaml` to `config/packages` directory. And travis will be green =)

**Comment by OskarStark** [comment]
I asked for a BlockBundle release @aLkRicha @covex-nn

**Comment by aLkRicha** [comment]
@OskarStark @covex-nn i'm waiting this PR https://github.com/sonata-project/SonataBlockBundle/pull/457

**Comment by covex-nn** [comment]
@OskarStark @aLkRicha since [3.10.0](https://github.com/sonata-project/SonataBlockBundle/releases/tag/3.10.0) there is no need to create a recipe for BlockBundle:

* Recipe is used only for new projects, it is used for initial configuration only
* Projects with BlockBundle < v3.10 are already configured
* New projects will use BlockBundle >= 3.10.0 with it's default configuration. See it for yourself:

    ```bash
    composer create-project symfony/skeleton .
    composer config extra.symfony.allow-contrib true
    composer require sonata-project/block-bundle
    ```

    There you will see error about `templating` service `The service "sonata.block.service.container" has a dependency on a non-existent service "templating"`. And this dependency must be removed.

But i can be wrong: may be we should create recipe for 3.9.*, what do you think?

**Comment by covex-nn** [comment]
@OskarStark @aLkRicha https://github.com/sonata-project/SonataBlockBundle/pull/457 is merged and released with [3.11.0](https://github.com/sonata-project/SonataBlockBundle/releases/tag/3.11.0), but i'm still not sure if we need a recipe for BlockBundle

**Comment by Nyholm** [comment]
So, should I close this?

**Comment by jordisala1991** [comment]
Default context is no longer required, so many we don't need this recipe

**Comment by jordisala1991** [comment]
And this one can be closed, it is not needed

**Pull request closed without merging**